### PR TITLE
Add Constant class.

### PR DESCRIPTION
**Pull request by radumereuta**
Continuation from #882 
Let's see what happens.

### CODE REVIEW

**Comment by radumereuta** [comment]
Jenkins: test this please

File: src/javasources/KTool/src/org/kframework/kil/Constant.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+    protected String value;
+
+    /**
+     * Represents a lexical production from the parser.
+     * Should be used only in the 'parser' package.
+     * @param sort Sort of the production.
+     * @param value String representation of the constant.
+     */
+    public Constant(Sort sort, String value) {
```
Comment (radumereuta) [inline src/javasources/KTool/src/org/kframework/kil/Constant.java]: I'm using this in JUnit tests. The code won't look too good if I delete this constructor.

**Comment by radumereuta** [comment]
Putting the code that flatten constants in `NormalizeAST` would be wrong. That's the job of `FlattenSyntax`.
At the moment the only place I'm creating Constants, is in the new parser, and I'm calling `FlattenSyntax` before the AST gets to the compilation stages.

**Comment by radumereuta** [comment]
I think I've addressed all the comments.
@dwightguth please review.

File: src/javasources/KTool/src/org/kframework/kil/Constant.java (hunk 1 above)
Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/kil/Constant.java]: It's not generally a good idea for production code to depend on the details of your tests. I'm more concerned about developers calling this constructor by mistake and then getting a NullPointerException than I am about what tests look like. The purpose of a test is to execute cleanly and efficiently, and test the right functionality, not to look pretty.

**Comment by dwightguth** [comment]
Okay. I am fine with FlattenSyntax then. We can always change it later if we run into problems.

**Comment by radumereuta** [comment]
I removed the constructor.
@dwightguth please review.

**Pull request merged**